checks/apps/quantumespresso/quantumespresso_check.py

```diff
@@ -1,43 +1,86 @@
-# Copyright 2016-2022 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# Copyright 2016-2023 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
 # ReFrame Project Developers. See the top-level LICENSE file for details.
 #
 # SPDX-License-Identifier: BSD-3-Clause
 
+import pathlib
+import re
+import sys
+
 import reframe as rfm
 import reframe.utility.sanity as sn
 
-
-class QuantumESPRESSOCheck(rfm.RunOnlyRegressionTest):
-    scale = parameter(['small', 'large'])
-    variant = parameter(['maint', 'prod'])
-    modules = ['QuantumESPRESSO']
-    executable = 'pw.x'
-    executable_opts = ['-in', 'ausurf.in', '-pd', '.true.']
-    extra_resources = {
-        'switches': {
-            'num_switches': 1
+sys.path.append(str(pathlib.Path(__file__).parent.parent.parent / 'mixins'))
+from extra_launcher_options import ExtraLauncherOptionsMixin
+from sarus_extra_launcher_options import SarusExtraLauncherOptionsMixin
+from cuda_visible_devices_all import CudaVisibleDevicesAllMixin
+
+qe_tests = {
+    # R:total number of MPI ranks, T:threads per rank, P:walltime
+    'Au-surf': {
+        'zen2': {
+            # zen2 cpu nodes: 2sockets, 8 numa, 16c/numa = 128c (no MT)
+            'energy_reference': -11427.09017218,
+            'performance_reference': [{'R': 32, 'T': 4, 'P': 102.0}]
+        },
+        'zen3': {
+            # A100 nodes: 1socket, 4 numa, 16c/numa = 64c (no MT)
+            'energy_reference': -11427.09017218,
+            'performance_reference': [{'R': 4, 'T': 16, 'P': 32.0}]
         }
     }
-    strict_check = False
-    maintainers = ['LM']
-    tags = {'scs'}
+}
+
```

Comment:
```suggestion


```

Reply:
I didn't get this one. Do you want to have two empty lines between end of dictionaty and beginning of class declaration?

Reply:
yes